src/topp/HighResPrecursorMassCorrector.cpp

```diff
@@ -275,6 +281,84 @@ class TOPPHiResPrecursorMassCorrector :
       return corrected_precursors;
     }
 
+    set<Size> correctToHighestIntensityMS1Peak(PeakMap & exp, double mz_tolerance, bool ppm, vector<double> & deltaMZs,vector<double> & mzs, vector<double> & rts)
+    {
+      set<Size> corrected_precursors;
+      // load experiment and extract precursors
+      vector<Precursor> precursors;  // precursor
+      vector<double> precursors_rt;  // RT of precursor MS2 spectrum
+      vector<Size> precursor_scan_index;
+      getPrecursors_(exp, precursors, precursors_rt, precursor_scan_index);
+
+      for (Size i = 0; i != precursors_rt.size(); ++i)
+      {
+        // get precursor rt
+        double rt = precursors_rt[i];
+
+        // get precursor MZ
+        double mz = precursors[i].getMZ();
+
+        // cout << rt << " " << mz << endl;
+
+        // get precursor spectrum
+        MSExperiment<Peak1D>::ConstIterator rt_it = exp.RTBegin(rt - 1e-8);
+
+        // store index of MS2 spectrum
+        UInt precursor_spectrum_idx = rt_it - exp.begin();
+
+        // get parent (MS1) of precursor spectrum
+        rt_it = exp.getPrecursorSpectrum(rt_it);
+
+        if (rt_it->getMSLevel() != 1)
+        {
+          LOG_WARN << "Error: no MS1 spectrum for this precursor" << endl;
+        }
+
+        MSSpectrum<Peak1D>::ConstIterator left = rt_it->MZBegin(mz - mz_tolerance);
+        MSSpectrum<Peak1D>::ConstIterator right = rt_it->MZEnd(mz + mz_tolerance);
+
+        // no MS1 precursor peak in +- tolerance window found
+        if (left == right || left->getMZ() > mz + mz_tolerance)
+        {
+          LOG_WARN << "Error: The method highest_intensity_peak failed.";
```

Comment:
don't know if we should print a warning for every PC here? maybe just count how often it was tried and how often it failed and output this information later?